scalacheck/src/main/scala/pathy/scalacheck/GenPath.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2014 - 2015 SlamData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pathy
+package scalacheck
+
+import Path._
+
+import org.scalacheck.Arbitrary
+import org.scalacheck.Gen
+
+import scalaz.Show
+import scalaz.syntax.show._
+
+/** Represents a `Path[B,T,S]` indexed by another type `A` which is used to
+  * generate path segments.
+  *
+  * i.e. For some type `A` such that Arbitrary[A] and Show[A],
+  * Arbitrary[GenPath[B,T,S,A]] will generate an arbitrary Path[B,T,S] where
+  * every path segment is formed by Arbitrary[A].shows.
+  */
+final case class GenPath[B,T,S,A](path: Path[B,T,S]) extends AnyVal
```

Comment:
I think it's generally better to name things for what they "are" instead of "context in which they are intended to be used".  If you don't like `ConstrainedPath` do you have an alternative that conveys what it is?